test/controllers/auth/AuthActionSpec.scala

```diff
@@ -66,30 +67,72 @@ class AuthActionSpec extends FreeSpec with MustMatchers with MockitoSugar with O
   }
 
   type AuthRetrievals =
-    Option[String] ~ Enrolments ~ Option[Credentials] ~ ConfidenceLevel ~ Option[UserName] ~ LoginTimes ~ Option[
-      TrustedHelper] ~ Option[String]
+    Option[String] ~ Option[AffinityGroup] ~ Enrolments ~ Option[Credentials] ~ Option[String] ~ ConfidenceLevel ~ Option[
+      UserName] ~ LoginTimes ~ Option[TrustedHelper] ~ Option[String]
 
+  val nino = Fixtures.fakeNino.nino
   val fakeCredentials = Credentials("foo", "bar")
+  val fakeCredentialStrength = CredentialStrength.strong
   val fakeConfidenceLevel = ConfidenceLevel.L200
   val fakeLoginTimes = LoginTimes(DateTime.now(), None)
 
   def fakeSaEnrolments(utr: String) = Set(Enrolment("IR-SA", Seq(EnrolmentIdentifier("UTR", utr)), "Activated"))
 
+  def retrievalResult(
+    nino: Option[String] = Some(nino.toString),
+    affinityGroup: AffinityGroup = Individual,
+    saEnrolments: Enrolments = Enrolments(Set.empty),
+    credentialStrength: String = CredentialStrength.strong,
+    confidenceLevel: ConfidenceLevel = ConfidenceLevel.L200,
+    trustedHelper: Option[TrustedHelper] = None): Future[AuthRetrievals] =
+    Future.successful(
+      nino ~ Some(affinityGroup) ~ saEnrolments ~ Some(fakeCredentials) ~ Some(credentialStrength) ~ confidenceLevel ~ None ~ fakeLoginTimes ~ trustedHelper ~ None
+    )
+
   "A user without a L200 confidence level must" - {
-    "be redirected to the IV uplift endpoint" in {
-      when(mockAuthConnector.authorise(any(), any())(any(), any()))
-        .thenReturn(Future.failed(InsufficientConfidenceLevel()))
-      val authAction = new AuthActionImpl(mockAuthConnector, app.configuration, configDecorator)
-      val controller = new Harness(authAction)
-      val result = controller.onPageLoad()(FakeRequest("GET", "/personal-account"))
-      status(result) mustBe SEE_OTHER
-      redirectLocation(result).get must endWith(
-        "/mdtp/uplift?origin=PERTAX&confidenceLevel=200&completionURL=%2Fpersonal-account%2Fidentity-check-complete%3FcontinueUrl%3D%252Fpersonal-account&failureURL=%2Fpersonal-account%2Fidentity-check-complete%3FcontinueUrl%3D%252Fpersonal-account")
+    "be redirected to the IV uplift endpoint when" - {
+      "the user is an Individual" in {
+        when(mockAuthConnector.authorise[AuthRetrievals](any(), any())(any(), any())) thenReturn retrievalResult(
+          confidenceLevel = ConfidenceLevel.L100)
+
+        val authAction = new AuthActionImpl(mockAuthConnector, app.configuration, configDecorator)
+        val controller = new Harness(authAction)
```

Comment:
This setup is used throughout a lot of tests. Can remove the duplication by encapsulating it in a function. Try creating a method on the Harness companion object with the following signature
```
def retrievals(affinityGroup: AffinityGroup, confidenceLevel: ConfidenceLevel): Harness = ???
```
depending on how the other tests are structured you could consider incorporating the `retrievalResult` function into this method and using that instead to create the controllers